heasarc_retrieve_pipeline/core.py

```diff
@@ -274,6 +265,83 @@ def retrieve_heasarc_data_by_source_name(
 
     return results
 
+@flow
+def retrieve_heasarc_data_by_obsid(
```

Comment:
Avoid to hardcode mission-specific flags. Better to use a generic argument for the flags, and have mission-specific functions manage them transparently